endpoint.go

```diff
@@ -804,9 +810,27 @@ func (ep *endpoint) assignAddressVersion(ipVer int, ipam ipamapi.Ipam) error {
 		return nil
 	}
 
+	// Validate preferred address
+	if prefAdd != nil {
+		for _, d := range ipInfo {
```

Comment:
Why do we need this extra iteration for validation? Why can't we do this validation in the iterator below which anyways had to iterate over `ipInfo`? Reduces some extra boiler plate code